### PR TITLE
Migrate to ghcr

### PR DESCRIPTION
**Pull request by francoisferrand**
(already reviewed as https://github.com/scality/cloudserver/pull/5555)

- Migrate to ghcr
- Upgrade actions
- Build pykmip image
- Use official docker build steps

Issue: CLDSRV-524

### CODE REVIEW

**Review by nicolas2bert — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -11,36 +11,59 @@ on:
...
+          password: ${{ github.token }}
+      - name: Build and push image
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: .
+          tags: |
+            ghcr.io/${{ github.repository }}:${{ github.event.inputs.tag }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```
Comment (ghost) [inline .github/workflows/release.yaml:58]: Do we want to have a scope here?

Comment (francoisferrand) [reply .github/workflows/release.yaml:58]: not really: this is the "main" image, so we can use the default scope.
as long as no scope is shared between different images, there is no issue.

**Review by BourgoisMickael — APPROVED** [review]

**Pull request merged**